pkg/gitlab/gitlab_client.go

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gitlab
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/xanzy/go-gitlab"
+)
+
+// Allow mocking for tests
+var NewGitlabClient func(accessToken string) (*GitlabClient, error) = newGitlabClient
+
+type GitlabClient struct {
+	client *gitlab.Client
+}
+
+func newGitlabClient(accessToken string) (*GitlabClient, error) {
+	glc := &GitlabClient{}
+	c, err := gitlab.NewClient(accessToken)
+	if err != nil {
+		return nil, err
+	}
+	glc.client = c
+
+	return glc, nil
+}
+
+func (c *GitlabClient) branchExist(projectPath, branchName string) (bool, error) {
+	_, resp, err := c.client.Branches.GetBranch(projectPath, branchName)
+	if err != nil {
+		if resp.StatusCode == 404 {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func (c *GitlabClient) createBranch(projectPath, branchName, baseBranchName string) error {
+	opts := &gitlab.CreateBranchOptions{
+		Branch: &branchName,
+		Ref:    &baseBranchName,
+	}
+	_, _, err := c.client.Branches.CreateBranch(projectPath, opts)
+	return err
+}
+
+func (c *GitlabClient) deleteBranch(projectPath, branchName string) error {
+	_, err := c.client.Branches.DeleteBranch(projectPath, branchName)
+	return err
+}
+
+func (c *GitlabClient) filesUpToDate(projectPath, branchName string, files []File) (bool, error) {
+	for _, file := range files {
+		opts := &gitlab.GetRawFileOptions{
+			Ref: &branchName,
+		}
+		fileContent, resp, err := c.client.RepositoryFiles.GetRawFile(projectPath, file.FullPath, opts)
+		if err != nil {
+			if resp.StatusCode != 404 {
+				return false, err
+			}
+			return false, nil
+		}
+		if !bytes.Equal(fileContent, file.Content) {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func (c *GitlabClient) commitFilesIntoBranch(projectPath, branchName, baseBranch, commitMessage, authorName, authorEmail string, files []File) error {
```

Comment:
What is the variable `baseBranch` being used for here?

Reply:
Leftovers after changing the implementations. Will delete that parameter.